### PR TITLE
Add Payment Service

### PR DESCRIPTION
**Pull request by Maple5201**
## Summary
Add Payment Service with auth shop and user service

### CODE REVIEW

File: PaymentService/PaymentService/Dockerfile
Comment (tadhgryan2012) [inline PaymentService/PaymentService/Dockerfile:1]: We are using java 21

File: PaymentService/PaymentService/HELP.md
Comment (tadhgryan2012) [inline PaymentService/PaymentService/HELP.md:1]: This does not need to be commited

File: PaymentService/PaymentService/build.gradle
Comment (tadhgryan2012) [inline PaymentService/PaymentService/build.gradle:1]: Need to add spotless for linting

File: PaymentService/PaymentService/docker-compose.yml
Comment (tadhgryan2012) [inline PaymentService/PaymentService/docker-compose.yml:1]: The ports can be injected from the config service.

File: PaymentService/PaymentService/gradle/wrapper/gradle-wrapper.jar
Comment (tadhgryan2012) [inline PaymentService/PaymentService/gradle/wrapper/gradle-wrapper.jar:1]: Does not need to be commited

File: PaymentService/PaymentService/gradle/wrapper/gradle-wrapper.properties
Comment (tadhgryan2012) [inline PaymentService/PaymentService/gradle/wrapper/gradle-wrapper.properties:1]: Does not need to be comitted

File: PaymentService/PaymentService/gradlew
Comment (tadhgryan2012) [inline PaymentService/PaymentService/gradlew:1]: Does not need to be committed.

File: PaymentService/PaymentService/gradlew.bat
Comment (tadhgryan2012) [inline PaymentService/PaymentService/gradlew.bat:1]: Does not need to be committed.

File: PaymentService/PaymentService/settings.gradle
Comment (tadhgryan2012) [inline PaymentService/PaymentService/settings.gradle:1]: Does not need to be committed.
But should add this service to the root settings.gradle

File: PaymentService/PaymentService/src/main/init.sql
Comment (tadhgryan2012) [inline PaymentService/PaymentService/src/main/init.sql:1]: Some of these tables to made in other services. Please read the other services up and ping these services if you need this data for your service.

**Review by tadhgryan2012 — CHANGES_REQUESTED** [review]
Overall good work, few changes are needed to align with the overall project structure.
Have not checked the endpoints yet but will do once these changes have been made.

**Review by tadhgryan2012 — APPROVED** [review]

**Review by Niomahony — APPROVED** [review]
Possibly the greatest PR of all time

**Review by tadhgryan2012 — APPROVED** [review]
lgtm :D

**Pull request merged**